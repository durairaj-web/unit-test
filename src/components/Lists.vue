<template>
  <ul>
    <li v-for="(list, index) in todoList " :key="index">{{ list }}</li>
    <slot :msg="msg"/>
  </ul>
  
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class Lists extends Vue {
  @Prop({ default: () => [] }) public todoList!: string[];
  public msg= '';

  mounted(): void{
    this.msg = "from child"
  }
}
</script>
