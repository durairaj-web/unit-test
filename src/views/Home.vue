<template>
  <div class="page-container">
    <h1>Todo List</h1>
    <section class="todo">
      <input
        type="text"
        name="newItem"
        v-model.trim="newItem"
        autocomplete="off"
        @keyup.enter="appendNewItem"
      />
      <button @click="appendNewItem">Add</button>
      <Lists class="todo-list" :todoList="todoList" v-slot="{msg}">
        <p>hello</p>
        {{msg}}
      </Lists>
    </section>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import Lists from "../components/Lists.vue";

@Component({
  components: {
    Lists
  },
})
export default class TodoList extends Vue {
  public todoList: string[] = [];
  public newItem = "";

  // life cycle
  mounted(): void {
    this.todoList = ["one", "two", "three"];
  }

  // event
  appendNewItem(): void {
      this.todoList.push(this.newItem);
      this.newItem = "";
    }
  }
</script>
