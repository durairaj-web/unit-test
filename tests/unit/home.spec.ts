import { mount, shallowMount } from "@vue/test-utils";
import Home from "../../src/views/Home.vue";
import Lists from "../../src/components/Lists.vue";

describe("Home.vue", () => {
  it("mount the Home compoenent", () => {
    const wrapper = mount(Home);
    expect(wrapper.find(".page-container").exists()).toBe(true);
  });

  it("match the Home component h1 title", async () => {
    const title = "Todo List";
    const wrapper = mount(Home);
    expect(wrapper.find("h1").text()).toMatch(title);
  })
  it("test the List component by adding a new list item", async () => {
    const parentWrapper = mount(Home);
    const todo = parentWrapper.vm.$data.todoList;

    const childWrapper = mount(Lists);
    await childWrapper.setProps({ todoList: [...todo, "Four"] });
    expect(childWrapper.html()).toMatchSnapshot();
  });

  it("test the add button", async () => {
    const wrapper = shallowMount(Home);
    await wrapper.setData({ newItem: "new todo" });
    wrapper.find('button').trigger('click');
    expect(wrapper.vm.$data.todoList).toContain("new todo");
    expect(wrapper.vm.$data.todoList.join(",")).toMatchSnapshot();
  });
});
