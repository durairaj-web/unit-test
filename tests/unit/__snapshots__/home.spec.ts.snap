// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue test the List component by adding a new list item 1`] = `
<ul>
  <li>one</li>
  <li>two</li>
  <li>three</li>
  <li>Four</li>
</ul>
`;

exports[`Home.vue test the add button 1`] = `"one,two,three,new todo"`;
